packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -73,3 +89,65 @@ Learn more about compiler configuration at https://usehardhat.com/configuration"
 
   return resolved;
 }
+
+/**
+ * Receives a parsed stack trace of a MODULE_NOT_FOUND error and checks
+ * if the error comes from a missing peer dependency from a hardhat plugin.
+ *
+ * If that's the case, it throws an error. Otherwise it does nothing.
+ */
+export function analyzeModuleNotFoundStackTrace(
+  stackTrace: stackTraceParser.StackFrame[],
+  configPath: string
+) {
+  const throwingFile = stackTrace
+    .filter((x) => x.file !== null)
+    .map((x) => x.file!)
+    .find((x) => path.isAbsolute(x));
```

Comment:
Are they always absolute paths?

Reply:
The ones I tested are, but I'm not really sure.

The way I thought about this is just: if something here is wrong, then this function won't do anything and the error will be re-thrown. I mean, one way or another we'll throw an error, it's just that one is more useful than the other. In the worst case where we have a false negative, the user will see that there's a missing package and will have a hint. Not ideal, but I think we can iterate over this.

Reply:
I like that reasoning 👌